src/network/networkd-dhcp-common.c

```diff
@@ -278,6 +278,66 @@ int config_parse_dhcp6_mud_url(
                 void *data,
                 void *userdata) {
 
+        char ***l = data;
+        int r;
+
+        assert(l);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                *l = strv_free(*l);
+                return 0;
+        }
+
+        for (;;) {
+                _cleanup_free_ char *w = NULL, *unescaped = NULL;
+
+                r = extract_first_word(&rvalue, &w, NULL, 0);
```

Comment:
Hmm, I think this should use `EXRACT_CUNESCAPE|EXTRACT_UNQUOTE` flags here, no? i.e. let the function do the C unescaping for us, and honour quotes, so that people can insert spaces and stuff